ObjectMapper/Core/Map.swift

```diff
@@ -62,13 +63,8 @@ public final class Map {
 		currentKey = key
 		keyIsNested = nested
 		
-		// check if a value exists for the current key
-		if nested == false {
-			currentValue = JSONDictionary[key]
```

Comment:
This check is important for performance reasons. We previously didn't include it and the valueFor function became a performance bottleneck because it was being called so often without any need.

Can you please revert this